src/Microsoft.Extensions.Caching.Memory/MemoryCacheOptions.cs

```diff
@@ -9,10 +9,41 @@ namespace Microsoft.Extensions.Caching.Memory
 {
     public class MemoryCacheOptions : IOptions<MemoryCacheOptions>
     {
+        private long? _sizeLimit;
+        private double _removalPercentageOnOvercapacityCompaction = 0.05;
+
         public ISystemClock Clock { get; set; }
 
         public TimeSpan ExpirationScanFrequency { get; set; } = TimeSpan.FromMinutes(1);
 
+        public long? SizeLimit
```

Comment:
Calculate a percentage? What are their inputs? "If my cache is over capacity I need to purge some data. How much data do I purge?" I'll argue that they'll be more likely to start with a percentage than a fixed value, "free up 5% capacity", which is much simpler to reason about than to say "how big is my total cache? 1gb? Free up 50mb".